### PR TITLE
(WIP) Add docker deployment

### PR DESCRIPTION
**Pull request by leplatrem**
- [x] Merge with n1k0 install docs
- [x] Use environment variables
- [x] Publish 1.0.dev image on docker.io

### CODE REVIEW

**Comment by elemoine** [comment]
Nice! Out of curiosity, what's the exact goal of using Docker? Is it for daybed developers to quickly have a working dev environment? Or for people who want to deploy daybed for an actual use?

Btw here's a interesting article that suggests using an init process in Docker containers: http://phusion.github.io/baseimage-docker/, which sounds wise to me.

**Comment by Natim** [comment]
@elemoine It is more for really fast user deployement for the next release we will release the docker image pre configured as well as a Debian package to make it really fast and easy to deploy.

File: conf/production.ini [hunk 1]
```diff
@@ -0,0 +1,53 @@
+[app:main]
+use = egg:daybed
+# Backend configuration
+daybed.backend = daybed.backends.couchdb.CouchDBBackend
+backend.db_host = http://couchdb:5984/
+backend.db_name = daybed
+persona.secret = THIS IS SUPER SECRET.
+persona.audiences = http://localhost:8000 http://0.0.0.0:8000 http://127.0.0.1:8000
```
Comment (Natim) [inline conf/production.ini]: The user needs to change that for production.

Comment (leplatrem) [inline conf/production.ini]: If we remove the Persona stuff, nothing to change in production !

**Comment by leplatrem** [comment]
Thanks Eric for getting in touch :)

I can see two main use-cases of Docker for Daybed : 
- Frontend developers who don't want to bother with python and stuff. Just get a local Daybed instance running ; 
- Easy deployment of Daybed instances in production, with the idea that every Mozilla service can easily be deployed at home :)

Regarding init in Docker, we've had this discussion for a while at Makina. The main disavantage is the lack of cron. But we can survive this by logging to stdout and delegate log handling to docker host daemon...

File: conf/production.ini (hunk 1 above)
Comment (Natim) [inline conf/production.ini]: We will see but in the documentation there is a way to override the conf so it works for me.

**Comment by Natim** [comment]
Why is it a WIP? What are you expecting to add?

File: conf/production.ini (hunk 1 above)
Comment (almet) [inline conf/production.ini]: @leplatrem it seems that we finally will not remote persona, see #150 for more background on the why.

Comment (almet) [inline conf/production.ini]: s/remote/remove

Comment (leplatrem) [inline conf/production.ini]: Ok, we don't remove Personna, then I'll use environment variables to set this up.

**Comment by leplatrem** [comment]
Not WIP anymore :)
Ready to review/merge

**Notes:**
UWSGI supports environment variables substitution in ini files for the `[uwsgi]` section (`$(VAR)` syntax) . But Paste config parser does not expand them at all in the other sections, so I did it in python with `os.path.expandvars` (`${VAR}` syntax).

File: daybed/__init__.py [hunk 2]
```diff
@@ -53,9 +54,17 @@ def get_user(request):
     return request.db.get_user(userid)
 
 
+def settings_expandvars(settings):
+    """Expands all environment variables in a settings dictionary.
+    """
+    return dict((key, os.path.expandvars(value))
+                for key, value in settings.iteritems())
```
Comment (Natim) [inline daybed/__init__.py]: Please use `six.items` instead of `iteritems` since it break Py3 compatibility

**Comment by leplatrem** [comment]
Done :)

**Pull request merged**